public/pages/DetectorResults/utils/utils.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import { Detector } from '../../../models/interfaces';
+import {
+  NO_FULL_SHINGLE_ERROR_MESSAGE,
+  NO_DATA_IN_WINDOW_ERROR_MESSAGE,
+  NO_RCF_MODEL_ERROR_MESSAGE,
+} from './constants';
+import { DETECTOR_STATE, SHINGLE_SIZE } from '../../../utils/constants';
+import { DETECTOR_INIT_FAILURES } from '../../../pages/DetectorDetail/utils/constants';
+import moment, { Moment } from 'moment';
+
+export const IS_INIT_OVERTIME_FIELD = 'isInitOvertime';
+export const INIT_DETAILS_FIELD = 'initDetails';
+export const INIT_ERROR_MESSAGE_FIELD = 'initErrorMessage';
+export const INIT_ACTION_ITEM_FIELD = 'initActionItem';
+
+export const getDetectorInitializationInfo = (detector: Detector) => {
+  const currentTime = moment();
+
+  let result = {
+    [IS_INIT_OVERTIME_FIELD]: false,
+    [INIT_DETAILS_FIELD]: {},
+  };
+  if (isDetectorInitOverTime(currentTime, detector)) {
+    result[IS_INIT_OVERTIME_FIELD] = true;
+    result[INIT_DETAILS_FIELD] = getInitOverTimeDetails(detector);
+  }
+  return result;
+};
+
+const isDetectorInitOverTime = (currentTime: Moment, detector: Detector) => {
+  return (
+    detector &&
+    detector.curState === DETECTOR_STATE.INIT &&
+    detector.initializationError &&
+    !detector.initializationError.includes(NO_RCF_MODEL_ERROR_MESSAGE) &&
```

Comment:
If no full shingle data, will the error message be `NO_RCF_MODEL_ERROR_MESSAGE `?

Reply:
In the normal case, usually after 6 interval, NO_RCF_MODEL_ERROR_MESSAGE will show up, and model training is started. This check here is to avoid breaking normal case.